entproto/extension.go

```diff
@@ -22,12 +22,70 @@ import (
 	"path/filepath"
 	"strings"
 
+	"entgo.io/ent/entc"
 	"entgo.io/ent/entc/gen"
 	"github.com/jhump/protoreflect/desc"
 	"github.com/jhump/protoreflect/desc/protoprint"
 	"go.uber.org/multierr"
 )
 
+// ExtensionOpt is an option for the entproto extension.
+type ExtensionOpt func(*Extension)
+
+// New returns a new Extension configured by opts.
+func New(opts ...ExtensionOpt) *Extension {
```

Comment:
Ditto: https://github.com/ent/contrib/blob/master/entgql/extension.go#L205
```suggestion
func NewExtension(opts ...ExtensionOpt) *Extension {
```

Reply:
You may also want to change the function to return an error `(*Extension, error)`. You don't need this now, but it will be a pain to change it in the future in case we need it.